### ISSUE TITLE
inheretance behavior of `profile` overrides child with parent config

### ISSUE THREAD

**Issue by dmarra** [Sceptre/sceptre#1366]
### Subject of the issue
When `profile` is supplied in my config.yaml, but my parent config.yaml doe snot have it specified, the parent overrides the child rather than the other way around

### Your environment
* sceptre v4.2.1
* python 3.10
* windows 10

### Steps to reproduce
Given a folder structure like so:
```
config/
|_ config.yaml
|_ /childconfig
    |_ sometemplate.yaml
    |_ config.yaml
templates/
|_/childconfig
  |_ sometemplate.yaml
```

Where `./config/config.yaml` contains:

```yaml
project_code: my_project
region: us-east-1
```

and `./config/childconfig/config.yaml` contains:

```yaml
project_code: my_project
region: us-east-1
profile: dev
```

NOTE: `sometemplate.yaml` can be anything; the failure point is before any resources are actually created

running `sceptre --debug launch childconfig` should reproduce the issue

### Expected behaviour
I expect the profile `dev` to be used, with an output similar to:
```sh
[2023-07-31 13:35:51] - No cloudformation client found, creating one...
[2023-07-31 13:35:51] - Getting Boto3 session
[2023-07-31 13:35:51] - No Boto3 session found, creating one...
[2023-07-31 13:35:51] - Using cli credentials...
[2023-07-31 13:35:52] - Using credential set from sso: {'AccessKeyId': '****************ABCD', 'SecretAccessKey': '************************************ABCD', 'Region': 'us-east-1'}
[2023-07-31 13:35:52] - Boto3 session created

```

Where my profile is an AWS SSO profile. I would expect that launching this command would be using my profile

### Actual behavior
I see something like the following:

```sh
[2023-07-31 13:59:15] - No cloudformation client found, creating one...
[2023-07-31 13:59:15] - Getting Boto3 session
[2023-07-31 13:59:15] - No Boto3 session found, creating one...
[2023-07-31 13:59:15] - Using cli credentials...
[2023-07-31 13:59:15] - Using credential set from shared-credentials-file: {'AccessKeyId': '****************WXYZ', 'SecretAccessKey': '************************************WXYZ', 'Region': 'us-east-1'}
[2023-07-31 13:59:15] - Boto3 session created
```

Notice that its `Using credential set from shared-credentials-file` rather than what was set in sso (and being used by my chosen profile)

### Additional Info

I am noticing in the debug output the order of operations it is taking:

```sh
[2023-07-31 13:35:45] - Reading in 'childconfig\config.yaml' files...
[2023-07-31 13:35:45] - Config: {'project_path': 'C:\\project', 'stack_group_path': 'childconfig', 'project_code': 'my_project', 'region': 'us-east-1', 'profile': 'dev'}[
[2023-07-31 13:59:13] - Reading in 'config.yaml' files...
[2023-07-31 13:59:13] - Config: {'project_path': ''C:\\project', 'stack_group_path': '', 'project_code': 'my_project', 'region': 'us-east-1'}
```

**NOTE:** Some log lines not shared, and some modified slightly to remove sensitive information; this is intended to give a gist of what I am seeing

Seems the parent is being read in _after_ the child. Not sure if this is just a curiosity of the logging, or if its a clue as to the cause.

**Comment by dmarra** [comment]
I am also noticing this inheritance issue related to `project_code` as well; in that the parent overrides the child.

**Comment by jfalkenstein** [comment]
Hi Dan, I'm not sure how what you're experiencing is happening. I can say that we have several projects that override project_code in child configs (among other settings). 

I would ask, however: Do you happen to have AWS environment variables set that might be directing your AWS environment to that profile (i.e. AWS_PROFILE environment variable, default profile on your aws config file, etc...)? Right now, if you explicitly specify a profile, Sceptre will use that. But if you do not specify a profile, Sceptre will use whatever your environment is set up to use. I can imagine scenarios where it may _seem_ like the parent is overriding the child, but your specific AWS environment settings might be interfering.

**Comment by dmarra** [comment]
Ok I have more information. My original reproduction case wasn't detailed enough. consider _this_:

```
config/
|_ config.yaml
|_ shared.yaml
|_ /childconfig
    |_ sometemplate.yaml
    |_ config.yaml
templates/
|_ shared.yaml
|_ sometemplate.yaml
```

What's happening is that the resource creation for `sometemplate.yaml` is having all the inheritance from the child config applied to the config from the parent (`/childconfig/config.yaml` vs `/config.yaml`); the child one wins. 

HOWEVER

shared.yaml does not have settings from `/childconfig/config.yaml` applied to `config.yaml`, even though my command specified to launch `childconfig`.

I suppose I might have a misunderstanding of how the inheritance works with StackGroup config. My expectation was that:

1. my `project_code` would be overridden for _all_ stacks created, since I specified `childconfig` as a launch param. 
2.  The profile set in `/childconfig/config.yaml` would override for the entire process, but since I have a global stack config for one of my stacks, it wants to use the profile defined in the parent for it instead of using the one profile for the entire process.

I was hoping to be able to do something like change the project_code and profile used for the _entire_ deployment based on the configuration I use at launch, not where the config exists in relation to where the template's config is defined. I wanted to keep the `shared.yaml` global, because all its doing is making a bucket, and its the same for every environment/configuration. 

TL;DR: seems like StackGroup config and Stack config have a bit of a coupling issue?

Am I perhaps approaching this the wrong way, or is this more of a feature request than anything?

**Comment by jfalkenstein** [comment]
Oh, what you've described is how Sceptre is supposed to work. `shared.yaml` shouldn't be affected at all by the StackConfig of `/childconfig/config.yaml`. Children don't change their aunts and uncles. 

I would recommend one of the following:
* Set the `profile` in the parent config like like this: `profile: {{var.profile}}` and then you can run `sceptre --var profile=dev launch ...` and everything will get that profile
* Don't use a profile on your Stack Configs at all and instead just set the `AWS_PROFILE` environment variable before running Sceptre.

**Comment by dmarra** [comment]
I suppose that makes sense, although it does feel a bit weird that something like `profile` can't be globally overridden. Now that I understand the behavior though, I was able to get around it via the following:

1. setting the profile via variable like you suggested, and using `default` in cases where i dont want a specific profile
2. using a Jinja2 conditional to control the `project_code` based on the existence of another variable

for anyone who finds this via a search, it item 2 looks something like:

```yaml
project_code: {% if var.owner is defined %} {{var.owner}}-some-service {% else %} someservice {% endif %}
```

Thanks for the help @jfalkenstein !
